adam/perception/perception_graph_nodes.py

```diff
@@ -81,13 +81,23 @@ class ObjectClusterNode(GraphNode):
     center_y: Optional[float] = attrib(validator=optional(instance_of(float)))
     std: Optional[float] = attrib(validator=optional(instance_of(float)))
 
+    def dot_label(self):
+        return (
+            f"ObjectClusterNode(cluster_id={self.cluster_id!r})\n"
+            f"(viewpoint_id={self.viewpoint_id}, "
+            f"center=({self.center_x:.2g}, {self.center_y:.2g}), std={self.std:.2g})"
+        )
+
 
 @attrs(frozen=True, slots=True, eq=False)
 class CategoricalNode(GraphNode):
     """A node representing a categorical value feature"""
 
     value: str = attrib(validator=instance_of(str))
 
+    def dot_label(self):
+        return f"CategoricalNode(label=?, value={self.value})"
```

Comment:
Note to self to ensure I update this when I rebase affordances on this PR or vice versa.

Aside: I wish we were using 3.8 so we could just write this as `f"CategoricalNode({label=}, {value=})"` but alas we're not

Reply:
Right? I tried to write it that way and PyCharm reminded me we don't have that feature yet. 🙃

Reply:
I'd love to upgrade to 3.8 but I think something we're using (Vistautils maybe?) was causing problems when I tried. When I updated to 3.7 I tried to just jump to 3.8 but wasn't able to.